### ISSUE TITLE
[NEW]: AI model open-ness

### ISSUE THREAD

**Issue by bact** [w3c/dpv#315]
### Specs

AI

### New Concept(s)

> Transferred from #82 and referred to some of the proposed concepts there (like `new:OpenWeights`).

Term: (used as unique ID) can be a classification like `ai:MOFClass1Model`, `ai:MOFClass2Model`, `MOFClass3Model`; or a set of components as criteria for classification - as proposed in #82 (`new:OpenArchitecture`, `new:ClosedArchitecture`, etc)

Alternatively, this can be modelled as a property `ai:hasOpennessLevel` for `ai:Model`.

Label: (human label) "Model Openness"

Definition: (definition or description of term)

Parent term: (existing term in DPV) `ai:Model`

Source: (url or reference)

- Model Openness Framework
  - paper - https://arxiv.org/abs/2403.13784
  - assessment tool - https://isitopen.ai/
  - blog - https://lfaidata.foundation/blog/2024/04/17/introducing-the-model-openness-framework-promoting-completeness-and-openness-for-reproducibility-transparency-and-usability-in-ai/

Usage note: (considerations when using the term e.g. scope, caution)

- A data subject may agree to allow the use of their data for AI training on a condition that the AI model must be Class I (Open Science) according to Model Openness Framework.

--

> below is a text copied from #82

On "open-ness", as discussed in https://harshp.com/dev/dpv/ai-training, the "Model Openness Framework" (see ref below) provides a classification of model openness:

- Class III - Open Model
- Class II - Open Tooling
- Class I - Open Science

Class III is least open, Class I is the most open.

DPV already provides some concepts that can be used to describe components within each Class.

Class III will be related to these concepts:

- `new:OpenArchitecture` / `new:ClosedArchitecture` (Model Architecture)
- `new:OpenWeights` / `new:ClosedWeights` / `new:PartiallyOpenWeights` (Model Parameters - final checkpoint)
- `tech:Documentation` (Technical Report, Evaluation Results, Model Card, Data Card)
  - also `eu-aiact:TechnicalDocumentation`

Class II will be related to these concepts:

- `new:OpenSource` / `new:ClosedSource`
- `ai:ValidationData` / `ai:TestingData` (Evaluation Data)

Class I will be related to these concepts:

- `tech:Documentation` (Research Paper)
- `new:OpenWeights` / `new:ClosedWeights` / `new:PartiallyOpenWeights` (Model Parameters - intermediate checkpoints)

**Comment by coolharsh55** [comment]
@DBlankvoort can you help here please by suggesting the concepts to represent spectrum of open-ness (completely open to completely closed) based on research in https://osai-index.eu/ ?

**Comment by DBlankvoort** [comment]
Hi,

To represent openness from our perspective, we would suggest the introduction of the additional concepts:
- Training Code (the general code framework used to train the model, possibly provided as a version-controlled repository), corresponding to Class II.
- Training Pipeline (a reproducible pipeline for replicating the training of the model), corresponding to Class I.
- Training Data (reference to the datasets used to train the model, possibly hosted by the model provider if not otherwise available), corresponding to Class I.

For the rest the existing concepts look good. The Open/Closed/PartiallyOpen Weights (with the primary distinguishing factor for PartiallyOpen presumably being gating of the weights behind e.g. a license) and the inclusion of the data card in `tech:Documentation` are particularly appreciated.

Inclusion of these additional concepts would allow for representing the spectrum of open-ness in a way which allows for distinguishing different existing definitions of OSAI, in particular also the definition of OSAI from the Open Atom Foundation in the PRC and the definition from the OSI in the USA. See also https://osai-index.eu/osai-definitions.

**Comment by bact** [comment]
Thank you @DBlankvoort so much for a very detailed response.

My initial idea is to probably use MOF components (for openness classification) as a middle layer to map each requirement in a regulation to an actual arfefact that will fulfill that requirement. Some of these artefacts may already well-represented by DPV or other vocabs. For training code and training data, I'm currently thinking about borrow them from software bill of materials (SBOM) specs.

In parallel to this inquiry in DPV, I proposed to SPDX to extend their ID types to cover ID schema for elements that related to authors, papers, technical whitepapers (https://github.com/spdx/spdx-3-model/pull/1187) and extend their license information to include information about appropriate type of content for the license (https://github.com/spdx/spdx-3-model/issues/1181) (SPDX License List https://github.com/spdx/license-list-data already contain information if a license is OSI-approved or FSF "free", but lacks information about appropriate content, as proposed by MOF). 

An initial MOF-SPDX mapping is proposed here: https://docs.google.com/spreadsheets/d/1WkSYTzOD1ufptGKvbCuFaXcqYolO9hiGiIIKPV3QV04/edit?gid=0#gid=0

I'm currently looking only at the Open Source AI Definition 1.0 from OSI. Will looking more into the effort from OpenAtom and European Open Source AI Definition, thank you again for the pointers.

**Comment by coolharsh55** [comment]
Thank you @bact @DBlankvoort ; based on the discussion I think modelling transparency as a `Status` would be a better choice as it can be changed. I do not think its a "type" of model as the concept is describing how much information the Provider has declared for it. The concepts would look like this:

1. `ai:ModelOpennessStatus` as the status indicating to what degree the model is 'open' based on the definitions and uses in MoF and OSAI
2. Specific statuses as `ai:ModelOpennessClassI` for models that fulfil the open science criteria, `ai:ModelOpennessClassII` for models that fulfil open tooling criteria, and `ai:ModelOpennessClassIII` for models that fulfil open model criteria.
3. Each status instance (above) should be further explained as needing specific information and the concepts used for this (e.g. documentation types, model weights, workflows or pipelines).
4. We will need specific hierarchy/taxonomy of these descriptive concepts, e.g. Architecture as a concept with open/closed variants, same for model weights. Declaring these as types would not be accurate as the type should not change just because a new documentation has been provided.

**Comment by bact** [comment]
This will be 2 parts:

1) The openness status itself, it will be generally like

    `ai:Model has:Status ai:ModelOpennessStatus`

     with `ai:ModelOpennessMOFClassI`, `ai:ModelOpennessMOFClassII`, `ai:ModelOpennessMOFClassIII` (as @coolharsh55 suggested, with small modification of putting `MOF` qualifier since we may have other classification of openness, per comments by @DBlankvoort above)

2) The assessment to get the status

    - Information needed to know to assess, for each MOF class is now partly mapped here https://docs.google.com/spreadsheets/d/1WkSYTzOD1ufptGKvbCuFaXcqYolO9hiGiIIKPV3QV04/edit?gid=0#gid=0 
    - This could be also be linked to AI Quality in another discussion.

**Comment by coolharsh55** [comment]
To confirm, we need two things: 1) list of concepts using which the openness is being determined, which we don't have AFAIK; and 2) statuses to model the openness itself.

Rationale: Using [MOF Specification](https://lfaidata.foundation/wp-content/uploads/sites/3/2025/01/05_White_paper_MOF_Specification.pdf) pg.3 there is a list of the specific criteria included in each class. Not all of them are enumerable, and there are multiple approaches possible e.g., for _weights_ we can have a property from `Model` to `Weights` like `hasWeights` using which we can assess if weights are declared or not. Alternatively, we can have a status like `WeightsAvailabilityStatus` with enumerations `WeightsAvailable` and `WeightsUnavailable`. My preference is to to have the properties but not the statuses where possible as otherwise its redundant. So we should make a separate list using pg.3 above as a proposal for what additional concepts/properties are needed.

For openness, we then have the following:

1. `ModelOpennessStatus`: status of a model representing its _openness_
2. Instances: `ModelStatusOpen` representing a model is open to some degree (i.e. one of the three MOF classes), and `ModelStatusClosed` indicating a model is closed completely and no details are avaialable (i.e. none of the three MOF classes)
3. Further instances expanding `ModelStatusOpen` as `ModelStatusOpenMOFClass{I,II,III}` using the MOF framework as source

- I prefer using `Open` rather than `Openness`.
- @DBlankvoort proposed concepts regarding training code, training data, etc. get added under item 1 (additional concepts/properties needed to determine openness)
- The broad upper concepts of open/close allow other categories of openness to be modelled in the future 

@bact @DBlankvoort any issues with above?

**Comment by DBlankvoort** [comment]
Nope, seems good!

**Comment by bact** [comment]
Looks good.

Also agreed on avoiding redundancy.

**Comment by bact** [comment]
Btw, how to use https://w3c.github.io/dpv/2.3/tech/#hasLicense and https://w3c.github.io/dpv/2.3/tech/#License ?

Does it possible to use [SPDX License ID](https://spdx.org/licenses/) for the license?

Open Source Initiative just recently adopt the SPDX License ID:
https://opensource.org/blog/osi-adopts-spdx-ids-for-license-urls


Can we use `dpv:hasLicense` with `xsd:string` of value like "Apache-2.0" or should it be an `xsd:anyURI` like "https://opensource.org/license/Apache-2.0" or it needs on be in form of `tech:License` class?

**Comment by coolharsh55** [comment]
@bact good questions, should we also put this discussin in the TECH document? (+1 from me)

> Btw, how to use https://w3c.github.io/dpv/2.3/tech/#hasLicense and https://w3c.github.io/dpv/2.3/tech/#License ?
> 
> Does it possible to use [SPDX License ID](https://spdx.org/licenses/) for the license?

Yes, I think so, e.g. `tech:hasLicense <SPDX id>`

> 
> Open Source Initiative just recently adopt the SPDX License ID: https://opensource.org/blog/osi-adopts-spdx-ids-for-license-urls
> 
> Can we use `dpv:hasLicense` with `xsd:string` of value like "Apache-2.0" or should it be an `xsd:anyURI` like "https://opensource.org/license/Apache-2.0" or it needs on be in form of `tech:License` class?

We use `rangeInclude` for properties to allow cases like this, so it is okay to do `tech:hasLicense "string"` and `tech:hasLicense <IRI>`. Ideally IRIs are better for interoperability but I can understand the need for simple strings in use-cases. 

For TECH, we can state that `tech:License` can be SPDX IDs but we don't enforce these strictly i.e. you are also free to use strings, and then show both within an example?
